asdf/asdf.py

```diff
@@ -214,8 +247,10 @@ def _check_extensions(self, tree, strict=False):
             return
 
         for extension in tree['history']['extensions']:
+            installed = next((e for e in self.extensions if e.class_name == extension.extension_class), None)
```

Comment:
I guess I would prefer a bit more verbose construction than using next() to provide the default None. But that's just me.

Reply:
This will be changing in a future PR anyway, since the condition gets too complex for one line.  I'll leave it alone for the moment.